### PR TITLE
feat(sumstat imputation): adding class for sumstat imputation

### PR DESCRIPTION
**Pull request by addramir**
This PR adds class for summary statistics imputation.

### CODE REVIEW

**Comment by ireneisdoomed** [comment]
I'll review this one

File: src/gentropy/method/sumstat_imputation.py [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+
+class sumstat_imputation:
+    """Implementation of RAISS summary statstics imputation model."""
+
+    @staticmethod
+    def raiss_model(
+        zt: np.ndarray,
+        sig_t: np.ndarray,
+        sig_i_t: np.ndarray,
+        lamb: float = 0.01,
```
[the remark this answers is not in the document]
Comment (addramir) [reply src/gentropy/method/sumstat_imputation.py:19]: I think the main default value goes from wrapper function - `ImputationLauncher` - https://gitlab.pasteur.fr/statistical-genetics/raiss/-/blob/master/raiss/imputation_launcher.py and it has default 0.01. So I would use 0.01. In any case, I think, 0.1 is too much for regularisation.

File: src/gentropy/method/sumstat_imputation.py [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+
+import numpy as np
+import scipy.linalg
+
+
+class sumstat_imputation:
+    """Implementation of RAISS summary statstics imputation model."""
+
+    @staticmethod
+    def raiss_model(
```
[the remark this answers is not in the document]
Comment (addramir) [reply src/gentropy/method/sumstat_imputation.py:15]: To be honest, it is not a RAISS model. The model is much older then the RAISS package, that is a nice wraper of old model (it has no name as far as I know). I will think about the name and insert formula in docs.

[the remark this answers is not in the document]
Comment (addramir) [reply src/gentropy/method/sumstat_imputation.py:15]: I will leave it as raiss model but will add comment and formula to the docs.

[the remark this answers is not in the document]
Comment (addramir) [reply src/gentropy/method/sumstat_imputation.py:15]: Done.

File: docs/python_api/methods/sumstat_imputation.md
Comment (ireneisdoomed) [inline docs/python_api/methods/sumstat_imputation.md:1]: This is great, thanks!

**Review by ireneisdoomed — APPROVED** [review]
Ty for taking my comments into consideration!

**Comment by addramir** [comment]
@ireneisdoomed I answered/commented all comments except this one:

> I'd consider making use of the class properties here. Given that raiss_model mostly relies on its own inputs for internal function calls, continuously passing and documenting these parameters is very repetitive. For example, rather than passing sig_t across multiple functions, consider initialising the SummaryStatisticsImputation class where sig_t as a class attribute so that you can use self.sig_t directly within class methods. I think this would streamline the code and improve readability.

In general I agree with it, but potentially the structure of the class could be changed at the stage of its integration with the finemapper function. So I would leave it as it is if it is fine, at least for now.

**Comment by ireneisdoomed** [comment]
Thanks again. Ready for merge!

**Pull request merged**